src/main/scala/ReleaseExtra.scala

```diff
@@ -225,7 +225,7 @@ object ReleaseStateTransformations {
   }
 
   lazy val publishArtifacts = ReleaseStep(
-    action = publishArtifactsAction,
+    action = runPublishArtifactsAction,
```

Comment:
I think I found the difference - there's a change for multi module projects, if you don't have `releaseSettings` (or at a minimum `publishArtifactsAction := publish.value` defined) in every module, it doesn't publish the modules.